frontend/packages/dev-console/src/components/import/validation-schema.ts

```diff
@@ -0,0 +1,249 @@
+import * as yup from 'yup';
+import * as _ from 'lodash';
+import { convertToBaseValue } from '@console/internal/components/utils';
+import { isInteger } from '../../utils/yup-validation-util';
+
+const hostnameRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/;
+const pathRegex = /^\/.*$/;
+const nameRegex = /^([a-z]([-a-z0-9]*[a-z0-9])?)*$/;
+const projectNameRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/;
+
+export const urlRegex = /^(((ssh|git|https?):\/\/[\w]+)|(git@[\w]+.[\w]+:))([\w\-._~/?#[\]!$&'()*+,;=])+$/;
+
+export const nameValidationSchema = yup
+  .string()
+  .matches(nameRegex, {
+    message:
+      'Name must consist of lower-case letters, numbers and hyphens. It must start with a letter and end with a letter or number.',
+    excludeEmptyString: true,
+  })
+  .max(253, 'Cannot be longer than 253 characters.')
+  .required('Required');
+
+export const projectNameValidationSchema = yup.object().shape({
+  name: yup
+    .string()
+    .matches(
+      projectNameRegex,
+      "Name must consist of lower case alphanumeric characters or '-' and must start and end with an alphanumeric character.",
+    )
+    .required('Required'),
+});
+
+export const applicationNameValidationSchema = yup.object().shape({
+  selectedKey: yup.string(),
+  name: yup.string(),
+});
```

Comment:
This will need to be coordinated with changes in https://github.com/openshift/console/pull/2815